yt/visualization/line_plot.py

```diff
@@ -0,0 +1,280 @@
+"""
+A mechanism for plotting field values along a line through a dataset
+
+
+
+"""
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2013, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+import numpy as np
+
+from collections import defaultdict
+from yt.funcs import \
+    iterable
+from yt.units.unit_object import \
+    Unit
+from yt.units.yt_array import \
+    YTArray
+from yt.visualization.base_plot_types import \
+    PlotMPL
+from yt.visualization.plot_container import \
+    PlotContainer, \
+    PlotDictionary, \
+    log_transform, \
+    linear_transform, \
+    invalidate_plot
+
+class LinePlotDictionary(PlotDictionary):
+    def __init__(self, data_source):
+        super(LinePlotDictionary, self).__init__(data_source)
+        self.known_dimensions = {}
+
+    def _sanitize_dimensions(self, item):
+        field = self.data_source._determine_fields(item)[0]
+        finfo = self.data_source.ds.field_info[field]
+        dimensions = Unit(
+            finfo.units, registry=self.data_source.ds.unit_registry).dimensions
+        if dimensions not in self.known_dimensions:
+            self.known_dimensions[dimensions] = item
+            ret_item = item
+        else:
+            ret_item = self.known_dimensions[dimensions]
+        return ret_item
+
+    def __getitem__(self, item):
+        ret_item = self._sanitize_dimensions(item)
+        return super(LinePlotDictionary, self).__getitem__(ret_item)
+
+    def __setitem__(self, item, value):
+        ret_item = self._sanitize_dimensions(item)
+        super(LinePlotDictionary, self).__setitem__(ret_item, value)
+
+    def __contains__(self, item):
+        ret_item = self._sanitize_dimensions(item)
+        return super(LinePlotDictionary, self).__contains__(ret_item)
+
+class LinePlot(PlotContainer):
+    r"""
+    A class for constructing line plots
+
+    Parameters
+    ----------
+
+    ds: :class:`yt.data_objects.static_output.Dataset`
+        This is the dataset object corresponding to the
+        simulation output to be plotted.
+    fields: string
+        The name(s) of the field(s) to be plotted.
+    start_point: n-element list, tuple, ndarray, or YTArray
+        Contains the coordinates of the first point for constructing the line.
+        Must contain n elements where n is the dimensionality of the dataset.
+    end_point: n-element list, tuple, ndarray, or YTArray
+        Contains the coordinates of the first point for constructing the line.
+        Must contain n elements where n is the dimensionality of the dataset.
+    figure_size: integer or two-element
+    resolution: int
+        How many points to sample between start_point and end_point for
+        constructing the line plot
```

Comment:
Could you add an example here?  If you make it a completely runnable (i.e., with imports, loading sample data), this can be run as a test.